pennylane_qiskit/qiskit_device.py

```diff
@@ -162,9 +162,27 @@ def __init__(self, wires, provider, backend, shots=1024, **kwargs):
             else:
                 raise ValueError("Backend {} does not support noisy simulations".format(backend))
 
+        # set transpile_args
+        self.transpile_args = {}
+        transpile_sig = inspect.signature(transpile).parameters
+        self.set_args(kwargs, transpile_sig, self.transpile_args)
+
         if "backend_options" in s.parameters:
             self.run_args["backend_options"] = kwargs
 
+    @staticmethod
+    def set_args(source, signature, destination):
+        """This static method is for fetching all signature matching parameters from source dict and appending them to destination dict.
```

Comment:
It might be worth describing the method in terms of its more general use rather than the specific case it is used for. All in all, it conditionally updates a dictionary with the content of another one. A minor suggestion on docstring descriptions, in general, is to apply a line break, once it gets too long.
```suggestion
        """This static method is for fetching selected parameters from the
        source dictionary and appending them to the destination dictionary.
```